### PR TITLE
ci: avoid race between bench and guide

### PR DESCRIPTION
**Pull request by davidhewitt**
The `main` guide build fails a lot, and I think it's because while the guide is building the benchmarks run and push their results to the `gh-pages` branch. I'm hoping by making the benchmark jobs run after the guide build, it'll fix this race.

### CODE REVIEW

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -103,3 +104,94 @@ jobs:
           publish_dir: ./public/
           full_commit_message: "Release ${{ needs.build.outputs.tag_name }}"
           keep_files: true
+
+  cargo-benchmark:
+    if: ${{ github.ref_name == 'main' }}
+    name: Cargo benchmark
+    needs: build
```
Comment (adamreichold) [inline .github/workflows/gh-pages.yml:111]: Isn't this called `guide-build` now?

Comment (davidhewitt) [reply .github/workflows/gh-pages.yml:111]: Yes, good spot.

**Comment by davidhewitt** [comment]
Not sure why the actions wouldn't start until I force-toggled the PR.

**Comment by adamreichold** [comment]
> Not sure why the actions wouldn't start until I force-toggled the PR.

GitHub still seems to have a bit of trouble with its mySQL1 cluster, c.f. https://www.githubstatus.com/#past-incidents and https://github.blog/2022-03-23-an-update-on-recent-service-disruptions/

**Review by messense — APPROVED** [review]

**Pull request merged**